qiskit_research/utils/dynamical_decoupling.py

```diff
@@ -86,18 +85,23 @@ def dynamical_decoupling_passes(
     Yields:
         Iterator[Iterable[BasePass]]: Transpiler passes used for adding DD sequences.
     """
-    durations = get_instruction_durations(backend)
-    pulse_alignment = backend.configuration().timing_constraints["pulse_alignment"]
+    for new_gate in [Xp, Xm, Yp, Ym]:
```

Comment:
To fix the mypy bug, maybe something like
```suggestion
    for new_gate in cast(List[Gate], [Xp, Xm, Yp, Ym]):
```